luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -0,0 +1,514 @@
+
+
+function gadget:GetInfo()
+	return {
+		name = "Team Power Watcher",
+        desc = "Tracks power of individual and total units per team. To be used for PvE dynamic difficulty and library functions to assertain aspects of game progression",
+		author = "SethDGamre",
+		date = "2024-07-12",
+		layer = -1,
+		enabled = true
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+local teamList = Spring.GetTeamList()
+local scavengerTeam
+local raptorTeam
+local aiTeams = {}
+local neutralTeam
+local humanTeams = {}
+local teamPowers = {}
+local peakTeamPowers = {}
+local unitsWithPower = {}
+local powerThresholds = {
+    {techLevel = 1, threshold = 9000},
+    {techLevel = 1.5, threshold = 45000},
+    {techLevel = 2, threshold = 90000},
+    {techLevel = 2.5, threshold = 230000},
+    {techLevel = 3, threshold = 350000},
+    {techLevel = 3.5, threshold = 475000},
+    {techLevel = 4, threshold = 600000},
+    {techLevel = 4.5, threshold = 725000}
+}
+
+for _, teamID in ipairs(teamList) do
+    Spring.Echo("teamID", teamID)
+    local teamLuaAI = Spring.GetTeamLuaAI(teamID)
+    if (teamLuaAI and string.find(teamLuaAI, "ScavengersAI")) then
+        scavengerTeam = teamID
+    elseif (teamLuaAI and string.find(teamLuaAI, "RaptorsAI")) then
+        raptorTeam = teamID
+    elseif select (4, Spring.GetTeamInfo(teamID, false)) then
+        aiTeams[teamID] = true
+    elseif teamID == tonumber(teamList[#teamList]) then
+        neutralTeam = teamID
+    else
+        humanTeams[teamID] = true
+    end
+end
+
+--assign team powers/peak powers to 0 to prevent nil
+for _, teamNumber in ipairs(teamList) do
+    teamPowers[teamNumber] = 0
+    peakTeamPowers[teamNumber] = 0
+end
+
+
+
+function gadget:UnitFinished(unitID, unitDefID, unitTeam)
+    unitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+    teamPowers[unitTeam] = (teamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+    if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
+        peakTeamPowers[unitTeam] = teamPowers[unitTeam]
+    end
+end
+
+function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
```

Comment:
If this uses `MetaUnitRemoved` instead, it will also cover `UnitTaken` i.e. a unit being captured or given away.